include/SocketWrapper/include/SocketWrapper.h

```diff
@@ -0,0 +1,54 @@
+#ifndef SOCKET_WRAPPER_H
+#define SOCKET_WRAPPER_H
+
+#include <boost/asio.hpp>
+#include <boost/asio/ssl.hpp>
+
+using TcpSocket = boost::asio::ip::tcp::socket;
+using TcpSocketPtr = std::shared_ptr<TcpSocket>;
+
+using SslSocket = boost::asio::ssl::stream<boost::asio::ip::tcp::socket>;
+using SslSocketPtr = std::shared_ptr<SslSocket>;
+
+class SocketWrapper
+{
+public:
+    explicit SocketWrapper(boost::asio::io_context io_context);
```

Comment:
Try passing io_context by reference. It should improve efficiency. It looks like you are using it in a way where only one global io_context should exist in the program